openpype/lib/applications.py

```diff
@@ -1506,7 +1507,11 @@ def _prepare_last_workfile(data, workdir):
     start_last_workfile = should_start_last_workfile(
         project_name, app.host_name, task_name, task_type
     )
-    data["start_last_workfile"] = start_last_workfile
+    if not data.get("start_last_workfile", True):
```

Comment:
You're not storing the result to `data` all the time and calling `should_start_last_workfile` even if it is not necessary.
```
start_last_workfile = data.get("start_last_workfile", True)
if start_last_workfile:
    start_last_workfile = should_start_last_workfile(
        project_name, app.host_name, task_name, task_type
    )
else:
    log.info("Explicitly forbidden to open last workfile")

data["start_last_workfile"] = start_last_workfile
```

Reply:
Ohh, you are right, this is much better.
Fixed.